py/jpy/src/main/c/jpy_jtype.c

```diff
@@ -862,7 +896,7 @@ int JType_ConvertPythonToJavaObject(JNIEnv* jenv, JPy_JType* type, PyObject* pyA
     } else if (PyBool_Check(pyArg) && (type == JPy_JObject || ((*jenv)->IsAssignableFrom(jenv, JPy_Boolean_JClass, type->classRef)))) {
         return JType_CreateJavaBooleanObject(jenv, type, pyArg, objectRef);
     } else if (JPy_IS_CLONG(pyArg) && (type == JPy_JObject || ((*jenv)->IsAssignableFrom(jenv, JPy_Integer_JClass, type->classRef)))) {
```

Comment:
`(*jenv)->IsAssignableFrom(jenv, JPy_Integer_JClass, type->classRef)` means:

"Can a java.lang.Integer be safely cast to classRef?"

If so, we could return a java.lang.Integer and know that a user expecting classRef would be able to safely use it.

Now, `JType_CreateJavaIntegerObjectEx` has a variable return type - and the same check we are doing doesn't exhaustively check that it's a safe return type.

At the very least, we need to add some more if-cases if we want to support JType_CreateJavaIntegerObjectEx - but, I'd suggest each of the internal cases you have in JType_CreateJavaIntegerObjectEx may need to be moved to individual if statements in this method.